### DISCUSSION TITLE
Cache Remix HTML using cloudflare cache API or upstash redis or cloudflare Kv

### DISCUSSION THREAD

**Discussion by omnaiduu** [remix-run/remix#8289] [General]
Hey I was trying to implement document caching in remix using upstash redis/cloudflare KV or workers cache API/ i could technically use http caching but the main benefit of using cache API or kV is that i can programmatically clear cache... So in server.entry before running remix and generate the response i can serve the response from kv or cache API.. if it's not present than we can put in cache.. but while putting the cache in api.. it's a blocking process and technically it should be wrapped inside a waituntill from context.....
I am not able to access the waituntill specifically in entry.server.. i tried using cloudflare workers and pages package and it's not working.. i can access the waituntill in loaders but that would add some 90ms roughly but documents caching always is faster.. can anyone please tell me is it just my mistake that I am not able to access the waituntill in entry.server.. you guys can access it.. i tired passing the context through getloadcontext.. it is not working

**Comment by candidia** [comment]
Should be able to access waitUntil like you described, eg https://github.com/kokeshing/my-remix-blog-example/blob/master/app/entry.server.tsx
